transmogrifier/sources/xml/dspace_dim.py

```diff
@@ -260,47 +183,174 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor] | None:
+        contributors = []
+        contributors.extend(list(cls._get_creators(source_record)))
+        contributors.extend(
+            list(cls._get_contributors_by_contributor_element(source_record))
+        )
+        return contributors or None
+
+    @classmethod
+    def _get_creators(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
+        for creator in source_record.find_all(
+            "dim:field", element="creator", string=True
+        ):
+            yield timdex.Contributor(
+                value=str(creator.string),
+                kind="Creator",
+            )
+
+    @classmethod
+    def _get_contributors_by_contributor_element(
+        cls, source_record: Tag
+    ) -> Iterator[timdex.Contributor]:
+        for contributor in source_record.find_all(
+            "dim:field", element="contributor", string=True
+        ):
+            yield timdex.Contributor(
+                value=str(contributor.string),
+                kind=contributor.get("qualifier") or "Not specified",
+            )
+
+    @classmethod
+    def get_dates(cls, source_record: Tag) -> list[timdex.Date] | None:
+        dates = []
+        for date in source_record.find_all("dim:field", element="date", string=True):
+            date_value = str(date.string.strip())
+            if validate_date(date_value, cls.get_source_record_id(source_record)):
+                if date.get("qualifier") == "issued":
+                    date_object = timdex.Date(value=date_value, kind="Publication date")
+                else:
+                    date_object = timdex.Date(
+                        value=date_value, kind=date.get("qualifier") or None
+                    )
+                dates.append(date_object)
+        dates.extend(list(cls._get_coverage_dates(source_record)))
+        return dates or None
+
+    @classmethod
+    def _get_coverage_dates(cls, source_record: Tag) -> Iterator[timdex.Date]:
+        for coverage_value in [
+            str(coverage.string)
+            for coverage in source_record.find_all(
+                "dim:field", element="coverage", qualifier="temporal", string=True
+            )
+        ]:
+            if "/" in coverage_value:
+                split = coverage_value.index("/")
+                gte_date = coverage_value[:split]
+                lte_date = coverage_value[split + 1 :]
+                if validate_date_range(
+                    gte_date,
+                    lte_date,
+                    cls.get_source_record_id(source_record),
+                ):
+                    yield timdex.Date(
+                        range=timdex.DateRange(
+                            gte=gte_date,
+                            lte=lte_date,
+                        ),
+                        kind="coverage",
+                    )
+            else:
+                yield timdex.Date(note=coverage_value, kind="coverage")
```

Comment:
Ah, more `yields`!  Maybe I just missed them in other transforms.  Digging this pattern.